docs/core/diagnostics/DiagnosticSource.csproj

```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">

```

Comment:
Can you put this and the .cs file into a subdirectory named "snippets"? That's the convention we use throughout this repo.